bot/interactions/commands/inventory/inventoryDispose.ts

```diff
@@ -0,0 +1,275 @@
+        data: {
+          type: InteractionType.InventoryDisposeToolDecline,
+          userDiscordId: interaction.user.id,
+          guildId,
+          payload: JSON.stringify({
+            walletId: context.data.walletId,
+            toolUniqueId: selectedTool.id,
+          } satisfies z.infer<typeof disposeContext>),
+        },
+      }),
+    ]);
+
+  const row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+    new ButtonBuilder()
+      .setCustomId(disposeInteractionAccept.id)
+      .setLabel("Accept")
+      .setStyle(ButtonStyle.Success),
+    new ButtonBuilder()
+      .setCustomId(disposeInteractionDecline.id)
+      .setLabel("Decline")
+      .setStyle(ButtonStyle.Danger),
+  );
+
+  await interaction.update({
+    content: sprintf(
+      "Are you sure, You want to dispose %s|%s",
+      toolData.emoji,
+      toolData.name,
+    ),
+    embeds: [],
+    components: [row],
+  });
+}
+
+export async function inventoryToolDisposeAccept(
+  interactionContext: InteractionContext,
+  interaction: AnyInteraction,
+) {
+  if (!interaction.isButton()) {
+    return void (await interaction.reply({
+      ephemeral: true,
+      content: "This interaction is an button",
+    }));
+  }
+
+  const context = disposeContext.safeParse(
+    JSON.parse(interactionContext.payload ?? "{}"),
+  );
+
+  if (!context.success) {
+    return await interaction.reply({
+      content: "Invalid interaction context",
+      ephemeral: true,
+    });
+  }
+
+  if (interaction.user.id !== interactionContext.userDiscordId) {
+    return await interaction.reply(
+      notYourInteraction(interactionContext, interaction),
+    );
+  }
+
+  const guild = interaction.guild;
+
+  if (!guild) {
+    return void (await interaction.reply({
+      ephemeral: true,
+      content: "This command is only available in servers.",
+    }));
+  }
+
+  if (guild.id !== interactionContext.guildId) {
+    return await interaction.reply(
+      wrongGuildForInteraction(interactionContext, interaction),
+    );
+  }
+
+  const toolUniqueId = z.string().safeParse(context.data.toolUniqueId);
+
+  if (!toolUniqueId.success) {
+    return await interaction.reply({
+      content: "Invalid value",
+      ephemeral: true,
+    });
+  }
+
+  const selectedTool = await prisma.shopItem.findUnique({
+    where: {
+      id: toolUniqueId.data,
+    },
+  });
+
+  if (!selectedTool) {
+    return await interaction.reply({
+      content: "Cannot find selected tool. Contact Developer.",
+      ephemeral: true,
+    });
+  }
+
+  const ToolType = (Object.keys(toolIds) as Array<ToolTypes>).find(
+    (key) => toolIds[key] === selectedTool.itemId,
+  ) as ToolTypes;
+  const toolData = buyToolItems[ToolType];
+
+  await prisma.shopItem.delete({
+    where: {
+      id: selectedTool.id,
+    },
+  });
+
+  await interaction.update(
+    await createEmbed(
+      interaction.user,
+      interaction.guild,
+      sprintf(
+        "Sucessfully disposed %s|%s from you inventory.",
```

Comment:
probably smart to make an abstraction "formatItem" in utils which does the formatting for items since it's used a lot and we likely want to change everywhere in the future.

```ts
function formatItem(type) {
   return sprintf("%s|%s", ...);
}